src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/integration/IntegrationTestBase.kt

```diff
@@ -1,16 +1,64 @@
 package uk.gov.justice.digital.hmpps.hmppsprobationestateapi.integration
 
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.TestInstance
+import org.mockserver.configuration.Configuration
+import org.mockserver.integration.ClientAndServer
+import org.mockserver.model.HttpRequest
+import org.mockserver.model.HttpResponse
+import org.mockserver.model.MediaType
+import org.slf4j.event.Level
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT
+import org.springframework.http.HttpHeaders
 import org.springframework.test.context.ActiveProfiles
 import org.springframework.test.web.reactive.server.WebTestClient
 
 @SpringBootTest(webEnvironment = RANDOM_PORT)
 @ActiveProfiles("test")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 abstract class IntegrationTestBase {
 
+  private var oauthMock: ClientAndServer =
```

Comment:
Interesting - I have not seen this `ClientAndServer` before - generally we use Wiremock directly

Reply:
I must admit I just followed suit with some of our other services which use mock server https://www.mock-server.com/ as the mocking mechanism